src/main/java/duke/save/SaveTaskListToText.java

```diff
@@ -0,0 +1,346 @@
+        }
+
+        return false;
+    }
+
+    // Solution below adapted from https://stackoverflow.com/questions/1625234/how-to-append-text-to-an-existing-file-in-java
+    public static boolean saveToDo(String taskName) {
+        String textToWrite1 = "t-/-0-/-" + taskName;
+        String textToWrite2 = "t-/-1-/-" + taskName;
+
+        try {
+            if (checkLineExist(textToWrite1) == false && checkLineExist(textToWrite2) == false) {
+                String dukeDirectory = "data/duke.txt";
+
+                File file = new File(dukeDirectory);
+                boolean fileIsEmpty = false;
+
+                if (file.length() == 0) {
+                    fileIsEmpty = true;
+                }
+
+                FileWriter addToDo = new FileWriter(dukeDirectory, true); //the true will append the new data
+                if (fileIsEmpty == false) {
+                    addToDo.write("\n");
+                }
+                addToDo.write(textToWrite1);
+                addToDo.close();
+            } else {
+                return false;
+            }
+        } catch(IOException ioe) {
+            System.err.println("IOException: " + ioe.getMessage());
+        }
+
+        return true;
+    }
+
+    // Solution below adapted from https://stackoverflow.com/questions/1625234/how-to-append-text-to-an-existing-file-in-java
+    public static boolean saveEvent(String taskName, String at) {
+        String eventToWrite1 = "e-/-0-/-" + taskName + "-/-/at" + at.substring(3);
+        String eventToWrite2 = "e-/-1-/-" + taskName + "-/-/at" + at.substring(3);
+
+        try {
+            if (checkLineExist(eventToWrite1) == false && checkLineExist(eventToWrite2) == false) {
+                String dukeDirectory = "data/duke.txt";
+                File file = new File(dukeDirectory);
+                boolean fileIsEmpty = false;
+
+                if (file.length() == 0) {
+                    fileIsEmpty = true;
+                }
+
+                FileWriter addToDo = new FileWriter(dukeDirectory, true); //the true will append the new data
+                if (fileIsEmpty == false) {
+                    addToDo.write("\n");
+                }
+                addToDo.write(eventToWrite1);
+                addToDo.close();
+            } else {
+                return false;
+            }
+        } catch(IOException ioe) {
+            System.err.println("IOException: " + ioe.getMessage());
+        }
+        return true;
+    }
+
+    // Solution below adapted from https://stackoverflow.com/questions/1625234/how-to-append-text-to-an-existing-file-in-java
+    public static boolean saveDeadline(String taskName, String by) {
+        String deadlineToWrite1 = "d-/-0-/-" + taskName + "-/-/by" + by.substring(3);
+        String deadlineToWrite2 = "d-/-1-/-" + taskName + "-/-/by" + by.substring(3);
+
+        try {
+            if (checkLineExist(deadlineToWrite1) == false && checkLineExist(deadlineToWrite2) == false) {
```

Comment:
This is a complex expression and it would help to break it down into multiple steps and declaring boolean variables. See guide on [complicated expressions](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions).